lib/rules/require-spaces-inside-parenthesized-expression.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Requires space after opening grouping round bracket and before closing.
+ *
+ * Types: `Boolean` or `Object`
+ *
+ * Values:
+ * - `true`: always require spaces inside grouping parentheses
+ * - `Object`:
+ *      - `"allExcept"`: `[ "{", "}", "function" ]` Ignore parenthesized objects and functions
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSpacesInsideParenthesizedExpression": true
+ *
+ * // or
+ *
+ * "requireSpacesInsideParenthesizedExpression": {
+ *     "allExcept": [ "{", "}" ]
+ * }
+ * ```
+ *
+ * ##### Valid for mode `true`
+ *
+ * ```js
+ * var x = ( 1 + obj.size ) * ( 2 );
+ * ```
+ *
+ * ##### Valid for mode `{ allExcept": [ "{", "}", "function" ] }`
+ *
+ * ```js
+ * var x = ( options || { x: true }).x;
+ * var global = (function() { return this; })();
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = (1 + obj.size) * (2);
+ * ```
+ */
+
+var assert = require('assert');
+var utils = require('../utils');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(value) {
+        var isObject = typeof value === 'object';
+
+        var error = this.getOptionName() + ' rule requires string value true or object';
+
+        if (isObject) {
+            assert('allExcept' in value, error);
+        } else {
+            assert(value === true, error);
+        }
+
+        this._exceptions = {};
+
+        if (isObject) {
+            (value.allExcept || []).forEach(function(value) {
+                this._exceptions[value] = true;
+            }, this);
+        }
+    },
+
+    getOptionName: function() {
+        return 'requireSpacesInsideParenthesizedExpression';
+    },
+
+    check: function(file, errors) {
+        var exceptions = this._exceptions;
+
+        file.iterateTokenByValue('(', function(token) {
+            var nextToken = file.getNextToken(token, {includeComments: true});
+            var value = nextToken.value;
+
+            // Skip empty parentheses and explicit exceptions
+            if (value === ')' || value in exceptions) {
```

Comment:
Because it's only an exception in one direction—if I universally except ")", then the immediately following close-parenthesis logic would erroneously allow "((…))".